cli/src/pcluster/config/config_patch.py

```diff
@@ -150,7 +150,7 @@ def _compare_list(self, base_section, target_section, param_path, data_key, fiel
 
         If update_key is not set we're considering Name as identifier.
         """
-        update_key = field_obj.metadata.get("update_key", "Name")
+        update_key = field_obj.metadata.get("update_key")
```

Comment:
We should add a unit tests to verify all the arrays defined in the schema have an update_key defined, to be sure this won't fail at runtime.

Reply:
Agreed... I think that these should be added to the `tests/pcluster/schemas` tests

Reply:
A note... I tried doing a `inspect.getmembers(pcluster.schemas, inspect.ismodule)` which should have returned the list of all submodules, however I checked that it only returned the first one.